sentry-android-ndk/src/main/jni/sentry.c

```diff
@@ -11,6 +11,190 @@ struct transport_options {
+        sentry_value_set_by_key(
+                crumb, "level", sentry_value_new_string(charLevel));
+    }
+
+    // overwrite timestamp that is already created on sentry_value_new_breadcrumb
+    if (timestamp) {
+        const char *charTimestamp = (*env)->GetStringUTFChars(env, timestamp, 0);
+        sentry_value_set_by_key(
+                crumb, "timestamp", sentry_value_new_string(charTimestamp));
+    }
+
+    // TODO: make this a method
+    if (data) {
+        sentry_value_t rvData = sentry_value_new_object();
+
+        jclass mapClass = (*env)->FindClass(env, "java/util/Map");
+        if (mapClass) {
+            jmethodID entrySet =
+                    (*env)->GetMethodID(env, mapClass, "entrySet", "()Ljava/util/Set;");
+            if (entrySet) {
+                jobject set = (*env)->CallObjectMethod(env, data, entrySet);
+
+                if (set) {
+                    jclass setClass = (*env)->FindClass(env, "java/util/Set");
+
+                    if (setClass) {
+                        jmethodID iterator =
+                                (*env)->GetMethodID(env, setClass, "iterator", "()Ljava/util/Iterator;");
+
+                        if (iterator) {
+                            jobject iter = (*env)->CallObjectMethod(env, set, iterator);
+
+                            if (iter) {
+                                jclass iteratorClass = (*env)->FindClass(env, "java/util/Iterator");
+
+                                if (iteratorClass) {
+                                    jmethodID hasNext = (*env)->GetMethodID(env, iteratorClass, "hasNext", "()Z");
+
+                                    if (hasNext) {
+                                        jmethodID next =
+                                                (*env)->GetMethodID(env, iteratorClass, "next", "()Ljava/lang/Object;");
+
+                                        if (next) {
+                                            jclass entryClass = (*env)->FindClass(env, "java/util/Map$Entry");
+                                            if (entryClass) {
+                                                jmethodID getKey =
+                                                        (*env)->GetMethodID(env, entryClass, "getKey", "()Ljava/lang/Object;");
+
+                                                jmethodID getValue =(*env)->GetMethodID(env, entryClass, "getValue", "()Ljava/lang/Object;");
+
+                                                // Iterate over the entry Set
+                                                while ((*env)->CallBooleanMethod(env, iter, hasNext)) {
+                                                    jobject entry = (*env)->CallObjectMethod(env, iter, next);
+                                                    jstring key = (jstring) (*env)->CallObjectMethod(env, entry, getKey);
+                                                    // TODO: value can be object e this would make it harder
```

Comment:
@bruno-garcia

Reply:
Ideally we'd just turn that into String, String in Java and then call into native.
I assume we serialize those, right? Then we'd do that in java before passing in to native.
That has the downside that we'll serialize stuff that might never even make it to the server.

I wonder how expensive all thsoe JNI calls are, + this serialization. All of that for crumbs data.

Maybe we leave that out on v1 of this feature?

Reply:
let's expose an option to turn this on via SentryOptions/Manifest